packages/backend/src/server/api/define.ts

```diff
@@ -48,42 +47,23 @@ export default function <T extends IEndpointMeta>(meta: T, cb: executor<T>)
 			id: '4267801e-70d1-416a-b011-4ee502885d8b',
 		}));
 
-		const [ps, pserr] = getParams(meta, params);
-		if (pserr) {
-			if (file) cleanup();
-			return Promise.reject(pserr);
-		}
+		if (validate) {
+			const valid = validate(params);
+			if (!valid) {
+				if (file) cleanup();
 
-		return cb(ps, user, token, file, cleanup);
-	};
-}
-
-function getParams<T extends IEndpointMeta>(defs: T, params: any): [Params<T>, ApiError | null] {
-	if (defs.params == null) return [params, null];
-
-	const x: any = {};
-	let err: ApiError | null = null;
-	Object.entries(defs.params).some(([k, def]) => {
-		const [v, e] = def.validator.get(params[k]);
-		if (e) {
-			err = new ApiError({
-				message: 'Invalid param.',
-				code: 'INVALID_PARAM',
-				id: '3d81ceae-475f-4600-b2a8-2bc116157532',
-			}, {
-				param: k,
-				reason: e.message,
-			});
-			return true;
-		} else {
-			if (v === undefined && Object.prototype.hasOwnProperty.call(def, 'default')) {
-				x[k] = def.default;
-			} else {
-				x[k] = v;
+				const err = new ApiError({
+					message: 'Invalid param.',
+					code: 'INVALID_PARAM',
+					id: '3d81ceae-475f-4600-b2a8-2bc116157532',
+				}, /* TODO {
+					param: k,
+					reason: e.message,
+				}*/);
```

Comment:
The documentation is at <https://ajv.js.org/api.html#error-parameters>. I think there could be multiple errors in which case we should only report the 1st for simplicity.
```suggestion
					param: validate.errors[0].propertyName,
					reason: validate.errors[0].message,
				});
```